borsh/src/schema_helpers.rs

```diff
@@ -9,7 +9,14 @@ use crate::{BorshDeserialize, BorshSchema, BorshSerialize};
 /// bytes describing the schema of the type. We deserialize this schema and verify that it is
 /// correct.
 pub fn try_from_slice_with_schema<T: BorshDeserialize + BorshSchema>(v: &[u8]) -> Result<T> {
-    let (schema, object) = <(BorshSchemaContainer, T)>::try_from_slice(v)?;
+    let mut v_mut = v;
+    let (schema, object) = <(BorshSchemaContainer, T)>::deserialize(&mut v_mut)?;
+    if !v_mut.is_empty() {
+        return Err(Error::new(
+            ErrorKind::InvalidData,
+            crate::de::ERROR_NOT_ALL_BYTES_READ,
+        ));
+    }
```

Comment:
I feel that we can use `from_slice` here, can't we?

Reply:
Yes, we can. Thanks!